src/models/document/document-content.ts

```diff
@@ -592,27 +592,28 @@ export const DocumentContentModel = types
     copyTilesIntoNewRows(tiles: IDragTileItem[], rowIndex: number) {
       const results: NewRowTileArray = [];
       if (tiles.length > 0) {
-        let rowDelta = 0;
+        let rowDelta = -1;
         let lastRowIndex = -1;
         tiles.forEach(tile => {
           let result: INewRowTile | undefined;
-          const content = safeJsonParse(tile.tileContent).content;
+          const parsedTile = safeJsonParse(tile.tileContent);
+          const content = parsedTile.content;
           if (content) {
+            if (tile.rowIndex !== lastRowIndex) {
+              rowDelta++;
+            }
             const rowOptions: INewTileOptions = {
```

Comment:
If you take my suggestion above, this should probably be renamed to `tileOptions`.